src/idpi/operators/hzerocl.py

```diff
@@ -34,13 +44,23 @@ def fhzerocl(t: xr.DataArray, hhl: xr.DataArray) -> xr.DataArray:
 
     # 3d field with values of height for those levels where temperature
     # is > 0 and it was < 0 on the level below. Otherwise values are NaN.
-    height2 = hfl.where((t >= t0) & (tkm1 < t0))
+    height0 = hfl.where((t >= t0) & (tkm1 < t0))
 
     # The previous condition can be satisfied on multiple levels.
     # Take the k indices of the maximum height value where the condition is satisfied
-    maxind: xr.DataArray = height2.fillna(-1).argmax(dim="z")  # type: ignore
+    maxind = cast(xr.DataArray, height0.fillna(-1).argmax(dim="z"))
+
+    if extrapolate:
+        # The full column is below freezing
+        below_ground = (t < t0).all("z")
+        # Temperature is increasing with decreasing altitude
+        positive_dt = t[{"z": -1}] - t[{"z": -2}] > 1e-10
+        # Allow t0 to be outside of the [height1, height2] range
+        cond = np.logical_and(below_ground, positive_dt)
+        maxind = maxind.where(~cond, -1)
```

Comment:
Does this mean that maxind = -1 when entire column is below freezing and temp is increasing with decreasing altitude? New to this syntax.